packages/amplify_authenticator/lib/src/widgets/buttons.dart

```diff
@@ -0,0 +1,217 @@
+import 'package:flutter/material.dart';
+
+import 'package:amplify_authenticator/src/constants/authenticator_constants.dart';
+import 'package:amplify_authenticator/src/constants/theme_constants.dart';
+import 'package:amplify_authenticator/src/state/inherited_auth_viewmodel.dart';
+
+//Reusable buttons
+
+class SignUpButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _authViewModel = InheritedAuthViewModel.of(context)!.signUpViewModel;
+    return AnimatedBuilder(
+        animation: _authViewModel,
+        builder: (context, child) {
+          Widget? _child;
+          var _callback;
+          if (_authViewModel.isBusy) {
+            _child = CircularProgressIndicator(color: Colors.white);
+            _callback = () {};
+          } else {
+            _child = Text("Sign Up");
+            _callback = _authViewModel.signUp;
+          }
+
+          return ElevatedButton(
+            key: Key("signUpButton"),
+            onPressed: _callback,
+            child: _child,
+            style: ButtonStyle(
+                padding:
+                    MaterialStateProperty.all<EdgeInsets>(EdgeInsets.all(10)),
+                backgroundColor: MaterialStateProperty.all<Color>(
+                  Theme.of(context).primaryColor != Colors.blue
+                      ? Theme.of(context).primaryColor
+                      : AuthenticatorColors.primary,
+                )),
+          );
+        });
+  }
+}
+
+class SignInButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _authViewModel = InheritedAuthViewModel.of(context)!.signInViewModel;
+
+    return AnimatedBuilder(
+        animation: _authViewModel,
+        builder: (context, child) {
+          Widget? _child;
+          var _callback;
+          if (_authViewModel.isBusy) {
+            _child = CircularProgressIndicator(color: Colors.white);
+            _callback = () {};
+          } else {
+            _child = Text("Sign In");
+            _callback = _authViewModel.signIn;
+          }
+
+          return ElevatedButton(
+            key: Key("signInButton"),
+            onPressed: _callback,
+            child: _child,
+            style: ButtonStyle(
+                padding:
+                    MaterialStateProperty.all<EdgeInsets>(EdgeInsets.all(10)),
+                backgroundColor: MaterialStateProperty.all<Color>(
+                  Theme.of(context).primaryColor != Colors.blue
+                      ? Theme.of(context).primaryColor
+                      : AuthenticatorColors.primary,
+                )),
+          );
+        });
+  }
+}
+
+class ConfirmButton extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final _authViewModel =
+        InheritedAuthViewModel.of(context)!.confirmSignUpViewModel;
+    return AnimatedBuilder(
```

Comment:
I think for now, I'm sticking with `AnimatedBuilder` in this PR but definitely considering in the next one. 
So to simplify the implementation, I might be going for  'InheritedNotifier'  instead of 'InheritedWidget' for the 'InheritedViewModel' widget.